Src/tim.c

```diff
@@ -52,12 +52,12 @@ void MX_TIM16_Init(void)
   TIM_OC_InitTypeDef sConfigOC;
   TIM_BreakDeadTimeConfigTypeDef sBreakDeadTimeConfig;
 
-  htim16.Instance = TIM16;
-  htim16.Init.Prescaler = 0;
+  htim16.Instance -> CCR3=936.5; //setting the compare register
+  htim16.Init.Prescaler = 1024; //brings freq down to 46875, need it at 50. So Compare match register 936.5 (+1) to get interrupt freq = 50Hz
```

Comment:
Can you explain what exactly you're trying to do here with CCR?

Reply:
I was trying to scale the frequency down, but decided against doing it that way and forgot to revert that.